File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    
    n = len(progresses)
    
    # 100에서 진행도를 빼고 남은 숫자를 보자
    doneDay = []
    for i in range(n):
        m = 100 - progresses[i]
        s = speeds[i]
        doneDay.append((m + s -1)//s)
    # print(doneDay)
    
    cnt = 1
    day = doneDay[0]
    
    for i in range(1, n):
        if doneDay[i] <= day:
            cnt += 1
        else:
            answer.append(cnt)
            cnt = 1
            day = doneDay[i]
    answer.append(cnt)
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;

public class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<int> answer = new List<int> {};
        
        int n = progresses.Length;
        int[] days = new int[n];
        
        for(int i = 0; i < n; i++){
            days[i] = (int)Math.Ceiling((100.0-progresses[i]) / speeds[i]);
            Console.WriteLine(days[i]);
        }
        int day = days[0];
        int cnt = 1;
        for(int j = 1; j < n; j++){
            if(day >= days[j]){
                cnt++;
            }
            else{
                answer.Add(cnt);
                day = days[j];
                cnt = 1;
            }
        }
        answer.Add(cnt);
        return answer.ToArray();
    }
}